Mathlib/Analysis/Convex/Cone/Pointed.lean

```diff
@@ -0,0 +1,200 @@
+/-
+Copyright (c) 2022 Apurva Nakade All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Apurva Nakade
+-/
+import Mathlib.Analysis.Convex.Cone.Dual
+import Mathlib.Algebra.Order.Nonneg.Ring
+import Mathlib.Algebra.Module.Submodule.Basic
+
+
+/-!
+# Pointed cones
+
+A *pointed cone* is defined to be a submodule of a module where the scalars are restricted to be
+nonnegative. This is equivalent to saying that as a set a pointed cone is convex cone which
+contains `0`. This is a bundled version of `ConvexCone.Pointed`. We choose the submodule definition
+as it allows us to use the `Module` API to work with convex cones.
+
+## TODO
+
+- Rewrite proper cones using pointed cones.
+- Construct products and/or direct sums of pointed cones.
+
+-/
+
+variable {𝕜 E F G : Type*}
+
+/-- A pointed cone is a submodule of a module with scalars restricted to being nonnegative. -/
+abbrev PointedCone (𝕜 : Type*) (E : Type*) [OrderedSemiring 𝕜] [AddCommMonoid E] [Module 𝕜 E] :=
+  have : Module { c : 𝕜 // 0 ≤ c } E := Module.compHom E (@Nonneg.coeRingHom 𝕜 _)
+  Submodule { c : 𝕜 // 0 ≤ c } E
+
+namespace PointedCone
+
+set_option quotPrecheck false in
+/-- The set of non-negative elements. -/
+scoped notation "𝕜≥0" => { c : 𝕜 // 0 ≤ c }
+
+section Definitions
+
+variable [OrderedSemiring 𝕜]
+variable [AddCommMonoid E] [Module 𝕜 E]
+
+/-- We consider the ambient space `E` as a module over just the non-negative scalars. -/
+instance : Module 𝕜≥0 E := Module.compHom E (@Nonneg.coeRingHom 𝕜 _)
```

Comment:
Can you move this instance to a much earlier file and make sure it's called `Nonneg.instModule`? If you can't find any sensible file, feel free to start writing `Algebra.Order.Nonneg.Smul` or `Algebra.Order.Nonneg.Module`.

Btw you're missing some lemmas:
```
@[simp, norm_cast] lemma Nonneg.coe_smul (a : 𝕜≥0) (x : E) : (a : 𝕜) • x = a • x := rfl
@[simp, norm_cast] lemma Nonneg.mk_smul (a) (ha) (x : E) : (⟨a, ha⟩ : 𝕜≥0) • x = a • x := rfl
```
+ some scalar tower instances

Reply:
Done.